addon/-private/system/model/internal-model.js

```diff
@@ -235,8 +241,68 @@ InternalModel.prototype = {
     return this.record;
   },
 
-  unloadRecord() {
+  directlyRelatedInternalModels() {
+    var array = [];
+    this.type.eachRelationship((key, relationship) => {
+      if (this._relationships.has(key)) {
+        var related = this._relationships.get(key).members.toArray();
+        array.push(...related);
```

Comment:
this may be fine, but there is an upper limit (in the low hundreds of thousands) that one can push onto an array.